test/passing/break_separators.ml

```diff
@@ -0,0 +1,174 @@
+  |] [@foo]
+
+(* this is a list *)
+let length =
+  ([ 0; 1; 2; 3; 4; 5; 6; 7; 8; 8; 9; 9; 10; 10; 11; 11; 12; 12; 12; 12; 13;
+     13; 13; 13; 14; 14; 14; (* foo *) 14; 15; 15; 15; 15; 16; 16; 16; 16;
+     16; 16; 16; 16; 17; 17; 17; 17 (* foo *); 17; 17; 17; 17; 18; 18; 18;
+     18; 18; 18; 18; 18; 19; 19; 19; 19; 19; 19; 19; 19; 20; 20; 20; 20; 20;
+     20; 20; 20; 20; 20; 20; 26; 26; 26; 26; 26; 27; 27; 27; 27;
+     2777777777777777777777777777777777; 27; 27; 27; 27; 27; 27; 27; 27; 27;
+     27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 28
+  ] [@foo])
+
+class
+  [ 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,
+    'bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb ]
+  x =
+  [ 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx,
+    'yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy ]
+  k
+
+type ( 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,
+       'bbbbbbbbbbbbbbbbbbbbbbbbbbbbb ) a =
+  ( 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,
+    'bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb )
+  e
+
+type ( 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,
+       'bbbbbbbbbbbbbbbbbbbbbbbbbbbbb ) a =
+  ('aaaaaaaaa, 'bbbbbbbbbbbb) e
+
+let ( xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx,
+      yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy,
+      zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz,
+      (aaaaaaaaaaaa, bbbbbbbbbbbb) ) =
+  ( ( xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx,
+      yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy,
+      zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz ),
+    (aaaaaaaaaaaaaa, bbbbbbbbbbbb) )
+
+type t = aaaaaaaaaaaa -> bbbbbbbbbbbb -> cccccccccc
+
+type t =
+  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa ->
+  bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb ->
+  ccccccccccccccccccccccccc
+
+type t =
+  (* foooooooooooo *)
+  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa ->
+  (* foooooooooooooooooooooooooooooooo*)
+  bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb ->
+  (* fooooooooooooooooo *)
+  ccccccccccccccccccccccccc ->
+  (* foooooo *)
+  foo * [`Foo of foo * foo] ->
+  ((* foooooooooooooooo *)
+  foo
+  * foo
+  * foo
+  * foo
+  * [ `Foo of
+      (* fooooooooooooooooooo *)
+      foo * foo * foo ->
+      foo ->
+      foo ->
+      (* foooooooooooo *)
+      foo ->
+      foo ->
+      foo * foo ->
+      foo * foo ->
+      foo * foo ]) ->
+  (* foooooooooooooooo *)
+  fooooooooooooooooo
+
+type t = {
+  (* fooooooooooooooooo *)
+    foo: foo;
+  (* foooooooooooooooooooooo fooooooooooooooooooo fooooooooooooooo
+     foooooooooooooooooo foooooooooooooooo *)
+    foo:
+      (* fooooooooooooooooooo *)
+      foooooooooooo ->
+      (* foooooooooooooo *)
+      foooooooooooooooo ->
+      foooooooooooooo ->
+      foooooooooo ->
+      fooooooooooooooo;
+  foo: foo;
+}
+
+[@@@ocamlformat "ocp-indent-compat"]
+
+type t =
+  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa ->
+  bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb ->
+  ccccccccccccccccccccccccc
+
+type t =
+  (string Location.loc * payload) list ->
+  ((string Location.loc * bool) list option
+   * (string Location.loc * payload) list) ->
```

Comment:
It seems odd to me to have `*` on the left and `->` on the right, but I guess that is intended.

Reply:
I am not sure it would more more readable if this operator was on the right.